INTER-224-ModalComponent-공용 컴포넌트 모달 구현

hoon730 opened the pull request:
### 반영 브랜치

INTER-224-ModalComponent -> dev

### PR 타입(하나 이상의 PR 타입을 선택해주세요)

- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트
- [ ] 코드 리펙토링

### 작업 내용

- 버튼 클릭 시, 모달 로직 구현
- default 및 custom 모달 구현
- Portal을 활용하여 DOM의 다른 계층 구조에서도 렌더링 가능

### 스크린샷 (선택)

![image](https://github.com/user-attachments/assets/b7b046f8-1a74-4178-ae27-7e4282aad1ce)

![image](https://github.com/user-attachments/assets/d8b3a03e-920b-48fe-a2bc-16c7decb0583)

### 리뷰 요구사항(선택)

- variants 속성이 불필요하거나 추가해야 될 것이 있으면 말씀해주세요!

hoon730 committed:
feat: ModalComponet variants 정의

-variants와 type정의

hoon730 committed:
feat: varaints 작성

- varaints작성 및 Modal.tsx

hoon730 committed:
feat: page생성

- modal페이지 출력 테스트

hoon730 committed:
feat: variants 추가 작성

-size 및 buttonLocation 추가

hoon730 committed:
style: variant 수정

-page 파일에 적용

hoon730 committed:
Merge branch 'dev' of https://github.com/Seoul-InterViewer/InterViewer into INTER-224-ModalComponent

hoon730 committed:
feat: modal, type 재정의

-motion 사용, type간소화

hoon730 committed:
refactor: page 변경

-className 프롭스 변경

hoon730 committed:
feat: Portal.tsx 생성

-Portal.tsx 정의 및 Modal.tsx에서 구현

hoon730 committed:
refactor: page 테스트

- default 모달, custom 모달 출력

hoon730 committed:
Merge branch 'dev' of https://github.com/Seoul-InterViewer/InterViewer into INTER-224-ModalComponent

hoon730 committed:
refactor: button.variants 이슈 #26

-variants 변경사항 merge 적용

GiToon10100011 commented on src/app/components/modal/Portal.tsx:
```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from "react";
+import { createPortal } from "react-dom";
+
+interface IPortalProps {
```
type.ts로 마이그레이션 부탁드립니다

GiToon10100011 commented on src/app/components/modal/index.ts:
```diff
@@ -0,0 +1,4 @@
+export * from "./Modal";
+export * from "./modal.type";
+export * from "./modal.variants";
+export * from "@/hooks/useModal";
```
전역 훅을 다시 재export 하는건 필요없어보입니다. 해당 부분 제거 부탁드리겠습니다.

GiToon10100011 commented on src/app/components/modal/modal.type.ts:
```diff
@@ -0,0 +1,8 @@
+export interface IModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  children?: React.ReactNode;
```
children을 옵셔널 타입으로 받은 이유가 있을까요?

GiToon10100011 commented on src/app/components/modal/page.tsx:
```diff
@@ -0,0 +1,105 @@
+      </div>
+
+      {/* 모달 컴포넌트가 띄워질 위치에  id="modal-root" 요소가 있어야 함 */}
+      <div id="modal-root">
```
매번 modal-root라는 아이디를 지닌 div를 추가하는게 아니라, rootLayout에 toast처럼 modal-root를 추가해놓으면, 훅을 통해 페이지 어디서든 쉽게 useModal만을 사용하고도 컨트롤이 가능할 것 같습니다.

GiToon10100011 replied:
layout 수정하시기 전에 dev와 미리 merge하시고 수정해주시길 바랍니다. 확인해보니 ToastContainer가 없어 충돌날 위험이 있습니다.

hoon730 committed:
Merge branch 'dev' of https://github.com/Seoul-InterViewer/InterViewer into INTER-224-ModalComponent

hoon730 committed:
refactor: index,Portal,tpye 리팩토링

-IPortalProps -> type.ts, index -> useModal export 제거, children 옵셔널 타입 제거, modal-root -> rootLayout으로 이동

GiToon10100011 commented:
LGTM!
